cognite/client/_api/transformations.py

```diff
@@ -0,0 +1,324 @@
+
+        Returns:
+            Union[Transformation, TransformationList]: Updated transformation(s)
+
+        Examples:
+
+            Update a transformation that you have fetched. This will perform a full update of the transformation::
+
+                >>> from cognite.experimental import CogniteClient
+                >>> c = CogniteClient()
+                >>> transformation = c.transformations.retrieve(id=1)
+                >>> transformation.query = "SELECT * FROM _cdf.assets"
+                >>> res = c.transformations.update(transformation)
+
+            Perform a partial update on a transformation, updating the query and making it private::
+
+                >>> from cognite.experimental import CogniteClient
+                >>> from cognite.experimental.data_classes import TransformationUpdate
+                >>> c = CogniteClient()
+                >>> my_update = TransformationUpdate(id=1).query.set("SELECT * FROM _cdf.assets").is_public.set(False)
+                >>> res = c.transformations.update(my_update)
+        """
+        return self._update_multiple(items=item)
+
+    def run(
+        self,
+        transformation_id: int = None,
+        transformation_external_id: str = None,
+        wait: bool = True,
+        timeout: Optional[float] = None,
+    ) -> TransformationJob:
+        """`Run a transformation. <https://docs.cognite.com/api/playground/#operation/runTransformation>`_
+
+        Args:
+            transformation_id (int): internal Transformation id
+            transformation_external_id (str): external Transformation id
+            wait (bool): Wait until the transformation run is finished. Defaults to True.
+            timeout (Optional[float]): maximum time (s) to wait, default is None (infinite time). Once the timeout is reached, it returns with the current status. Won't have any effect if wait is False.
+
+        Returns:
+            Created transformation job
+
+        Examples:
+
+            Run transformation to completion by id:
+
+                >>> from cognite.experimental import CogniteClient
+                >>> c = CogniteClient()
+                >>>
+                >>> res = c.transformations.run(id = 1)
+
+            Start running transformation by id:
+
+                >>> from cognite.experimental import CogniteClient
+                >>> c = CogniteClient()
+                >>>
+                >>> res = c.transformations.run(id = 1, wait = False)
+        """
+        utils._auxiliary.assert_exactly_one_of_id_or_external_id(transformation_id, transformation_external_id)
+
+        if transformation_external_id:
+            transformation_id = self.retrieve(external_id=transformation_external_id).id
+
+        response = self._post(
+            url_path=utils._auxiliary.interpolate_and_url_encode(
+                self._RESOURCE_PATH + "/{}/run", str(transformation_id)
```

Comment:
this is interesting, running a transformation only takes internal id, so we need to make two calls, one to get internal id if an ext id is present and other to actually run the transformation. Is it possible to run transformation via external Id itself?

Reply:
good catch! that's already implemented on the api (it wasn't when we first wrote this), so the double call isn't necessary